ball.h

```diff
@@ -8,11 +8,13 @@ namespace pong
 class Ball : public GameObject
 {
 private:
-    float mSpeed { 0.0f };
+    float mXSpeed { 0.0f };
+    float mYSpeed { 0.0f };
 
 public:
     void OnStart() override;
     void OnUpdate() override;
+    void OnCollisionStart(GameObject& other) override;
```

Comment:
I think it is nicer to let it be non-const. My reasoning is that sometimes it is nicer to do both operations on a collision between gameobjects from one side of the collision. For example a collision between some player/projectile would call TakeDamage() on the player and Destroy() on the projectile. Instead of calling each from their own respective OnCollisionStart() functions, the player only needs to implement the collision function and can call projectile.Destroy() as well. Sometimes it is just quicker to know what the behavior of a collision between two objects should be.